### PR TITLE
[Documentation] inline documentation added along with inline questions for more detailed inline documentation information.

### PR DESCRIPTION
**Pull request by rsmets**

### CODE REVIEW

**Comment by rsmets** [comment]
@jacobleesinger probably souldn't merge the PR with the the inline questions there. If you want you can just comment on the pr and I'll update the source or feel free to update the the source directly and push to this branch. Either way.

File: src/createKeyPairs.ts [hunk 1]
```diff
@@ -2,6 +2,10 @@ import { generateEccKeyPair, generateRsaKeyPair } from 'library-crypto-typescrip
 
 import { KeyPairSet } from './types';
 
+/**
+ * Question: Create key pair set?  token vs key pair set?
+ * @param encoding
+ */
```
Comment (jacobleesinger) [inline src/createKeyPairs.ts:8]: This is poorly named. It should be something like `createKeyPairSet`

File: src/createProof.ts [hunk 2]
```diff
@@ -2,6 +2,14 @@ import { sign } from 'library-crypto-typescript';
 
 import { Proof, JSONObj } from './types';
 
+/**
+ * Create cryptographic proof.
+ * Question: examples? use case?
```
Comment (jacobleesinger) [inline src/createProof.ts:7]: I think the examples/use cases would be the uses in the issuer and verifier apps. I suppose we could add something here as well though.

File: src/getKeyFromDIDDoc.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 import { PublicKeyInfo, DidDocument } from './types';
 
+// Question: What type of DID documents would have secp256r1 vs RSA keys?
```
Comment (jacobleesinger) [inline src/getKeyFromDIDDoc.ts:3]: secp256r1 keys are used for signing/verification and RSA keys are used for encryption. Currently all DID documents have secp256r1 keys and only holder DID Documents have RSA keys

File: src/helper.ts [hunk 4]
```diff
@@ -7,10 +7,24 @@ export const getUUID = (): string => {
...
+ * @param publicKeyObj
+ * @param data
+ */
 export const doEncrypt = (did: string, publicKeyObj: PublicKeyInfo, data: JSONObj): EncryptedData => {
   return (encrypt(did + '#' + publicKeyObj.id, publicKeyObj.publicKey, data, publicKeyObj.encoding));
 };
 
+/**
+ * Verify the signature on the provided data object.
+ * Question: 'pem' = 'pem' ? Needs some clarification.
```
Comment (jacobleesinger) [inline src/helper.ts:22]: `'base58' | 'pem'` is the argument type, the second `'pem'` is the default value

File: src/types.d.ts [hunk 5]
```diff
@@ -1,8 +1,16 @@
...
+ */
 export interface KeyPair {
   privateKey: string;
   publicKey: string;
 }
 
+/**
+ * Collection of key pairs. One for singing purposes and the other for encryption.
+ * The signing key pair is generally an ecc key while the encryption is RSA.
+ * Question:(ish) please verify my above assertion.
```
Comment (jacobleesinger) [inline src/types.d.ts:12]: yup

File: src/types.d.ts [hunk 6]
```diff
@@ -38,6 +55,10 @@ export interface PublicKeyInfo {
   status: 'valid' | 'invalid';
 }
 
+/**
+ * Type to encapsulate information related to a credential proof.
+ * Question: credential proof? is that correct?
```
Comment (jacobleesinger) [inline src/types.d.ts:60]: Proof for any signed object, which includes Credentials, Presentations, and PresentationRequests

File: src/types.d.ts [hunk 7]
```diff
@@ -101,6 +139,10 @@ export interface UnsignedCredential {
   expirationDate?: Date;
 }
 
+/**
+ * Type which incorporates the relevant credential information in addition to a cryptographic proof.
+ * Question: can assert the credential is "verified"..? Maybe wrong verbiage.
```
Comment (jacobleesinger) [inline src/types.d.ts:144]: "verifiable", not "verified", and yes.

File: src/types.d.ts [hunk 8]
```diff
@@ -59,13 +83,21 @@ export interface DidDocument {
   }[]
 }
 
+/**
+ * Type to encapsulate an encrypted key.
+ * Question: I think this needs more info... it is unclear to me just looking at the name wether this is a key for encryption or if it really is an "encrypted key"
+ */
```
Comment (jacobleesinger) [inline src/types.d.ts:89]: It's actually both. We can't encrypt credentials directly with RSA keys (they're too large), so instead we create a  symmetric AES key, encrypt the credential with the AES key, and then encrypt the AES key with the RSA key to share with the holder

File: src/types.d.ts [hunk 9]
```diff
@@ -59,13 +83,21 @@ export interface DidDocument {
...
   iv: string;
   key: string;
   algorithm: string;
   did: string;
 }
 
+/**
+ * Type to encapsulate information along side its encryption key.
+ * Question: this one also seems a little odd... the the data encrypted already with the key?
+ */
```
Comment (jacobleesinger) [inline src/types.d.ts:100]: see above. the data is encrypted with the key encapsulated here, which is itself encrypted with the RSA key from a DID document

File: src/restApiHelper.ts [hunk 10]
```diff
@@ -3,6 +3,10 @@ import fetch from 'node-fetch';
 import { JSONObj, RESTData, RESTResponse } from './types';
 import { CustError } from './error';
 
+/**
+ * Helper to handle network requests.
+ * @param inputObj
+ */
```
Comment (jacobleesinger) [inline src/restApiHelper.ts:9]: I really don't like this name, there's nothing inherently RESTful about it

File: src/helper.ts [hunk 11]
```diff
@@ -7,10 +7,24 @@ export const getUUID = (): string => {
   return (uuidv4());
 };
 
+/**
+ * Encrypt the provided did string.
+ * @param did
+ * @param publicKeyObj
+ * @param data
+ */
```
Comment (jacobleesinger) [inline src/helper.ts:15]: It's not encrypting the did, it's encrypting the `data`.

File: src/types.d.ts (hunk 8 above)
Comment (rsmets) [reply src/types.d.ts:89]: Ah cool got it, just like tls - cool.

File: src/createKeyPairs.ts (hunk 1 above)
Comment (rsmets) [reply src/createKeyPairs.ts:8]: Going to hold off on this until we touch base with Liam regarding some other potential name changes.

Comment (rsmets) [reply src/createKeyPairs.ts:8]: Updated. Will need to do in down stream projects when bumping the version.

File: src/restApiHelper.ts (hunk 10 above)
Comment (rsmets) [reply src/restApiHelper.ts:9]: updated the function to be `makeNetworkRequest` and the file to be `networkRequestHelper`

File: src/types.d.ts [hunk 12]
```diff
@@ -82,11 +112,17 @@ export interface Claim {
   [name: string]: ClaimValue;
 }
 
+/**
+ * Type to tie an id attribute to a claim used in the context of subject's credentials.
```
Comment (jacobleesinger) [inline src/types.d.ts:116]: Can be one or more claims (technically zero or more) `[key: string]: value;` is typescript's syntax for a string index, meaning the object can have any arbitrary string keys with values of the matching type

File: src/types.d.ts [hunk 13]
```diff
@@ -59,13 +81,21 @@ export interface DidDocument {
...
+ */
 export interface EncryptedKey {
   iv: string;
   key: string;
   algorithm: string;
   did: string;
 }
 
+/**
+ * Type to encapsulate encrypted information along side its encrypted, decryption key.
```
Comment (jacobleesinger) [inline src/types.d.ts:96]: no comma

File: src/types.d.ts [hunk 14]
```diff
@@ -1,8 +1,15 @@
+/**
+ * Public and Private corresponding key pair
+ */
 export interface KeyPair {
   privateKey: string;
   publicKey: string;
 }
 
+/**
+ * Collection of key pairs. One for singing purposes and the other for encryption.
```
Comment (jacobleesinger) [inline src/types.d.ts:10]: ~singing~ signing

File: src/helper.ts (hunk 11 above)
Comment (jacobleesinger) [reply src/helper.ts:15]: conversation was marked resolved but the comment wasn't updated

File: src/getDIDDoc.ts [hunk 15]
```diff
@@ -1,7 +1,13 @@
 import { CustError } from './error';
-import { makeRESTCall } from './restApiHelper';
+import { makeNetworkRequest } from './networkRequestHelper';
 import { RESTData, RESTResponse, DidDocument } from './types';
 
+/**
+ * Get a Did document from the did and url provided.
```
Comment (jacobleesinger) [inline src/getDIDDoc.ts:6]: are we stylizing did as `did`, `Did` or `DID` in documentation? It should be consistent regardless. (question for Liam, probably)

File: src/helper.ts (hunk 11 above)
Comment (rsmets) [reply src/helper.ts:15]: Oops! good catch.

File: src/types.d.ts (hunk 12 above)
Comment (rsmets) [reply src/types.d.ts:116]: Ahh I see. Cool.

Comment (rsmets) [reply src/types.d.ts:116]: So `claimName` in the follow code snippet is just for human readability? 
```

export interface CredentialSubject {
  id: string;
  [claimName: string]: ClaimValue;
}```

File: src/getDIDDoc.ts (hunk 15 above)
Comment (rsmets) [reply src/getDIDDoc.ts:6]: Hm yeah. I am indifferent.

**Comment by rsmets** [comment]
Hmm, I think I'll just refrain from marking your comments as resolved, Jacob. Will just acknowledge with an emoji, update the code and let you resolve them on the next pass.

File: src/types.d.ts (hunk 12 above)
Comment (jacobleesinger) [reply src/types.d.ts]: exactly, `claimName` could be anything there

File: src/types.d.ts [hunk 16]
```diff
@@ -59,34 +81,67 @@ export interface DidDocument {
...
 export type ClaimList = ClaimValue[];
+
+/**
+ * Type to encapsulate valid ClaimValue types.
+ */
 export type ClaimValue = ClaimList | ClaimPrimitive | ClaimDict;
+
+/**
+ * Interface to encapsulate an arbitrary number (0 to n) of string keys with values of type ClaimValue.
+ * Question: This seems like a duplicate of the Claim type below? Also doesn't look like this is used anywhere. See like can be deleted and ClaimValue type above (line 106) can be updated accordingly? Or is this the same of Claim type on purpose and just not used yet.
```
Comment (jacobleesinger) [inline src/types.d.ts:121]: I guess it is, yeah. Btw all of these `Claim` types are the result of typescript not having a built-in type for "anything that can be serialized as JSON"

Comment (rsmets) [reply src/types.d.ts]: I see. I kind of like the explicitness of them.

**Pull request merged**